Haptics/app/src/main/java/com/example/android/haptics/samples/ui/views/HomeRoute.kt

```diff
@@ -15,16 +15,199 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Text
+import androidx.compose.material.TextButton
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.platform.LocalView
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import com.example.android.haptics.samples.R
+import com.example.android.haptics.samples.ui.theme.HapticSamplerTheme
+import com.example.android.haptics.samples.ui.theme.buttonSurface
+import com.example.android.haptics.samples.ui.theme.onButtonSurface
+import com.example.android.haptics.samples.ui.theme.subtitleVariant
+import kotlinx.coroutines.launch
 
 @Composable
-fun HomeRoute() {
-    // TODO(jmylen): Implement this view in next commit.
-    Text(stringResource(R.string.home_screen_title), modifier = Modifier.padding(start = 32.dp))
+fun HomeRoute(
+    viewModel: HomeViewModel,
+) {
+    HomeScreen(
+        homeViewModelState = viewModel.viewModelState, onHapticFeedbackButtonClicked = viewModel::onHapticFeedbackButtonClicked,
+        onSnackbarMessage = viewModel::onSnackbarMessage, scrollState = viewModel.scrollState,
+    )
+}
+
+@Composable
+private fun HomeScreen(
+    homeViewModelState: HomeViewModelState,
+    onHapticFeedbackButtonClicked: (view: View, hapticFeedbackType: HapticFeedbackCategoryType, effect: Int) -> Unit,
+    onSnackbarMessage: (message: String) -> Unit,
+    scrollState: ScrollState = rememberScrollState(),
+) {
+    val coroutineScope = rememberCoroutineScope()
+    Column(
+        Modifier
+            .padding(horizontal = 16.dp)
+            .verticalScroll(scrollState),
+        horizontalAlignment = Alignment.Start
+
+    ) {
+        Text(
+            text = stringResource(R.string.home_screen_title),
+            style = MaterialTheme.typography.h4,
+            modifier = Modifier.padding(top = 36.dp, bottom = 16.dp)
+        )
+
+        for (category in homeViewModelState.hapticFeedbackCategories) {
+            HapticFeedbackCategory(label = category.label) {
+
+                // 2 buttons for ever row for each haptic feedback category.
+                for (buttons in category.buttons.chunked(2)) {
+                    Row(
+                        Modifier
+                            .padding(bottom = 16.dp)
+                            .fillMaxWidth(),
+                        horizontalArrangement = Arrangement.SpaceBetween
+                    ) {
+                        for ((index, button) in buttons.withIndex()) {
+                            // Add some spacing before the second button.
```

Comment:
Does it make sense to have a smaller 4dp padding around every button instead?

Reply:
The reason I'm doing it like this is because in the mocks the buttons align with the label where if I add padding to the button it shifts the button row to the right. I might be able to do it another way, just thought this was the easiest to visualize.